src/cmd.rs

```diff
@@ -125,13 +126,60 @@ impl<'a, T: FromRedisValue + 'a> AsyncIter<'a, T> {
 }
 
 #[cfg(feature = "aio")]
-impl<'a, T: FromRedisValue + Unpin + 'a> Stream for AsyncIter<'a, T> {
+impl<'a, T: FromRedisValue + 'a> AsyncIter<'a, T> {
+    /// ```rust,no_run
+    /// # use redis::AsyncCommands;
+    /// # async fn scan_set() -> redis::RedisResult<()> {
+    /// # let client = redis::Client::open("redis://127.0.0.1/")?;
+    /// # let mut con = client.get_async_connection().await?;
+    /// con.sadd("my_set", 42i32).await?;
+    /// con.sadd("my_set", 43i32).await?;
+    /// let mut iter: redis::AsyncIter<i32> = con.sscan("my_set").await?;
+    /// while let Some(element) = iter.next_item().await {
+    ///     assert!(element == 42 || element == 43);
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub async fn next_item(&mut self) -> Option<T> {
+        let inner = if let IterOrFuture::Iter(ref mut inner) = self.inner {
+            inner
+        } else {
+            panic!("next_item shouldn't be called in the middle of poll_next");
```

Comment:
This doesn't seem like a good idea. Futures (and streams) have no guarantees w.r.t being polled to completion, and are expected to work regardless (at least most of the time).
If this method must stay, then we could probably implement it in terms of `StreamExt::next` and (possibly) look into deprecating it.

Reply:
Oh, that's right! I can just use `StreamExt::next` on self here.. I don't know how I missed that, thanks!. About the deprecation, I don't know who should make that decision, but imho makes sense to just expose the api as an Stream than a custom async iterator